.github/workflows/check-reference.yml

```diff
@@ -0,0 +1,34 @@
+name: Check patch references
+
+on:
+  push:
+    branches-ignore:
+      - dependabot/**
+  schedule:
+    # Once every day at midnight UTC
+    - cron: "0 0 * * *"
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout patches
+        uses: actions/checkout@v4
+
+      - name: Checkout core
+        uses: actions/checkout@v4
+        with:
+          repository: 'homebrew/homebrew-core'
+          path: homebrew-core
+
+      - name: Detect references
+        run: |
+          patches=$(find -name "*.patch" -printf "%f\n")
+          status=0
+          for patch in $patches; do
+            if ! git -C "homebrew-core" grep -r "${patch/./}" > /dev/null; then
+              echo "$patch not needed!"
+              status=1
+            fi
+          done
```

Comment:
Maybe:

```shell
#! /bin/bash

core_patches=$(git -C "$(brew --repo homebrew/core)" grep -h "Homebrew/formula-patches" | awk -F/ '{print $NF}' | tr -d '"')
formula_patches_here=$(git ls-files | grep -v "LICENSE" | grep -v "README.md" | grep -v ".github/workflows/triage-issues.yml")
status=0
for patch in $formula_patches_here; do
  if ! grep -q "$patch" <<< "$core_patches"; then
    echo "Unused patch: $patch."
    status=1
  fi
done
exit $status
```

Reply:
```shell
    echo "Unused patch: $patch; deleting."
    rm "$patch"
```

And then the GitHub Actions add and push changes stuff we have in other workflows?

Reply:
Wait, no, I got these the wrong way around. 😩

Reply:
It's actually:

```shell
#!/bin/bash

core_patches=$(git -C "$(brew --repo homebrew/core)" grep -h "Homebrew/formula-patches" | awk -F/ '{print $NF}' | tr -d '"')
all_patches=$(git ls-files | grep -v "LICENSE" | grep -v "README.md" | grep -v ".github/workflows/triage-issues.yml")
status=0

for patch in $all_patches; do
# If patch is not in core patches, it is unused.
  if ! grep -q "$core_patches" <<< "$patch"; then
    echo "Unused patch: $patch."
    status=1
  fi
done
exit $status
```

Reply:
I pushed a commit - hope you don't mind!